docs/develop/understanding-the-javascript-api-for-office.md

```diff
@@ -28,31 +28,89 @@ For more details around the Office.js CDN, including how versioning and backward
 
 **Applies to:** All add-in types
 
-Office.js provides an initialization event which gets fired when the API is fully loaded and ready to begin interacting with the user. You can use the **initialize** event handler to implement common add-in initialization scenarios, such as prompting the user to select some cells in Excel, and then inserting a chart initialized with those selected values. You can also use the initialize event handler to initialize other custom logic for your add-in, such as establishing bindings, prompting for default add-in settings values, and so on.
+Your code must not call any Office.js APIs until the library is fully loaded. There are two events that your code can handle to ensure that the library is loaded. They are described in the sections below. We recommend that you use the newer, more flexible, `Office.onReady`. But the older `Office.initialize` is still supported. The main differences between them:
 
-At a minimum, the initialize event would look like the follow example:     
+- You can assign only one handler to `Office.initialize`. But you can assign multiple handlers, in different places in your code, to `Office.onReady`.
+
+- The `Office.initialize` event fires only once: at the end of the internal process in which Office.js initializes itself. And it fires immediately after the internal process ends. If the code in which you assign a handler to the event executes too long after the event fires, then your handler doesn't run. As a practical matter, this means that you must assign the handler as virtually the first thing your custom script does. The `Office.onReady` event fires when `Office.initialize` does but it also fires, in effect, whenever another handler is assigned to it. For example, you could have a handler with custom initialization logic assigned to `Office.onReady` as soon as your custom script loads (in, for example, an Immediately Invoked Function Expression). But you could also have a button in the task pane, whose script assigns another handler to `Office.onReady`. If so, the second handler runs when the button is clicked.
+
+- The `Office.onReady` event has a default handler that returns a Promise object. This gives you an alternative way of responding to the event. See **Initialize with Office.onReady** below.
```

Comment:
I don't think this comes out clearly.  It's more of "Office.onReady can be used with a Promise pattern or a callback pattern -- you choose"